webodf/tests/gui/ZoomHelperTests.js

```diff
@@ -0,0 +1,238 @@
+        t.scrollPane.style.overflow = "auto";
+    };
+
+    this.tearDown = function () {
+        t = {};
+        core.UnitTest.cleanupTestAreaDiv();
+    };
+
+    function createInnerScrollPane(width, height) {
+        var innerScrollPane = document.createElement("div");
+        innerScrollPane.id = "innerScrollPane";
+        innerScrollPane.style.transformOrigin = "top left";
+        innerScrollPane.style.width = width + "px";
+        innerScrollPane.style.height = height + "px";
+        innerScrollPane.style.position = "relative";
+        innerScrollPane.style.overflow = "auto";
+        innerScrollPane.style.border = "1px solid green";
+        return innerScrollPane;
+    }
+
+    function createSimulatedContent(width, height) {
+        var simulatedContent = document.createElement("div");
+        simulatedContent.id = "simulatedContent";
+        simulatedContent.style.transformOrigin = "top left";
+        simulatedContent.style.width = width + "px";
+        simulatedContent.style.height = height + "px";
+        simulatedContent.style.position = "relative";
+        simulatedContent.style.overflow = "hidden";
+        simulatedContent.style.border = "1px solid black";
+        for(var i = 0; i < 1000; i++) {
+            simulatedContent.appendChild(t.document.createTextNode(i + "_" + i + " "));
+        }
+        return simulatedContent
+    }
+
+    function createRedSquare(top, left) {
+        var redSquare = document.createElement("div");
+        redSquare.id = "redSquare";
+        redSquare.style.position = "absolute";
+        redSquare.style.width = "5px";
+        redSquare.style.height = "5px";
+        redSquare.style.top = top + "px";
+        redSquare.style.left = left + "px";
+        redSquare.style.backgroundColor = "red";
+        return redSquare;
+    }
+
+    function setZoomLevel_Maintains_scroll_position_on_zoom_in() {
+        t.scrollPane.style.width = "100px";
+        t.scrollPane.style.height = "100px";
+        var simulatedContent = createSimulatedContent(200, 200);
+        t.scrollPane.appendChild(simulatedContent);
+        t.zoomHelper.setZoomableElement(simulatedContent);
+        simulatedContent.appendChild(createRedSquare(49, 80));
+
+        t.scrollPane.scrollTop = 50;
+        t.scrollPane.scrollLeft = 80;
+
+        var zoom = 2;
+        t.zoomHelper.setZoomLevel(zoom);
+
+        r.shouldBe(t, "t.scrollPane.scrollTop", "100", SCROLL_POSITION_ABSOLUTE_TOLERANCE);
+        r.shouldBe(t, "t.scrollPane.scrollLeft", "160", SCROLL_POSITION_ABSOLUTE_TOLERANCE);
+        r.shouldBe(t, "redSquare.getBoundingClientRect().left", "t.scrollPane.getBoundingClientRect().left", SCROLL_POSITION_ABSOLUTE_TOLERANCE);
+        r.shouldBe(t, "redSquare.getBoundingClientRect().top", "t.scrollPane.getBoundingClientRect().top", SCROLL_POSITION_ABSOLUTE_TOLERANCE);
+    }
+
+    function setZoomLevel_Maintains_scroll_position_on_zoom_out() {
+        t.scrollPane.style.width = "100px";
```

Comment:
Ah like have it default as that and the ones which need a different width set it there?